qa/integration-tests/src/test/java/io/zeebe/broker/it/network/Ipv6IntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+          .enableIpv6(true)
+          .build();
+
+  private List<ZeebeBrokerContainer> containers;
+  private List<String> initialContactPoints;
+  private ZeebeGatewayContainer gateway;
+
+  @Before
+  public void setup() {
+    initialContactPoints = new ArrayList<>();
+    containers =
+        IntStream.range(0, CLUSTER_SIZE)
+            .mapToObj(i -> new ZeebeBrokerContainer(ZEEBE_IMAGE_VERSION))
+            .collect(Collectors.toList());
+
+    gateway = new ZeebeGatewayContainer(ZEEBE_IMAGE_VERSION);
+    IntStream.range(0, CLUSTER_SIZE).forEach(i -> configureBrokerContainer(i, containers));
+    configureGatewayContainer(gateway, initialContactPoints.get(0));
+  }
+
+  @After
+  public void tearDown() {
+    containers.parallelStream().forEach(GenericContainer::stop);
+  }
+
+  @Test
+  public void shouldCommunicateOverIpv6() {
+    // given
+    containers.parallelStream().forEach(GenericContainer::start);
+    gateway.start();
+
+    // when
+    final ZeebeClientBuilder zeebeClientBuilder =
+        ZeebeClient.newClientBuilder()
+            .usePlaintext()
+            .gatewayAddress(gateway.getExternalGatewayAddress());
+    try (final var client = zeebeClientBuilder.build()) {
+      final Topology topology = client.newTopologyRequest().send().join(5, TimeUnit.SECONDS);
+      // then - can find each other
+      TopologyAssert.assertThat(topology).isComplete(3, 1);
+    }
+  }
+
+  private void configureBrokerContainer(final int index, final List<ZeebeBrokerContainer> brokers) {
+    final int clusterSize = brokers.size();
+    final var broker = brokers.get(index);
+    final var hostNameWithoutBraces = String.format("%s%d", BASE_PART_OF_SUBNET, index + 2);
+    final var hostName = String.format("[%s]", hostNameWithoutBraces);
+
+    initialContactPoints.add(hostName + ":" + ZeebePort.INTERNAL.getPort());
+
+    broker
+        .withNetwork(network)
+        .withNetworkAliases(NETWORK_ALIAS)
+        .withCreateContainerCmdModifier(
+            createContainerCmd ->
+                createContainerCmd
+                    .withIpv6Address(hostNameWithoutBraces)
+                    .withHostName(hostNameWithoutBraces))
+        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
+        .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
+        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
+        .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(REPLICATION_FACTOR))
+        .withEnv("ZEEBE_BROKER_CLUSTER_PARTITIONCOUNT", String.valueOf(PARTITION_COUNT))
+        .withEnv(
+            "ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", String.join(",", initialContactPoints))
+        .withEnv("ZEEBE_BROKER_NETWORK_COMMANDAPI_ADVERTISEDHOST", hostNameWithoutBraces)
+        .withEnv("ZEEBE_BROKER_NETWORK_INTERNALAPI_ADVERTISEDHOST", hostNameWithoutBraces)
+        .withEnv("ZEEBE_BROKER_NETWORK_MONITORINGAPI_ADVERTISEDHOST", hostNameWithoutBraces)
```

Comment:
```suggestion
        .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISEDHOST", hostNameWithoutBraces)
```

This will be picked up by the various APIs.